tests/specifications/googlefonts_test.py

```diff
@@ -135,50 +135,118 @@ def test_example_checkrunner_based(cabin_regular_path):
       break
 
 
+class CheckResultsEvaluation():
+
+  def __init__(self, log_messages):
+    self.log_messages = log_messages
+
+  def assert_result(self, expected_status, expected_message=None):
+    last_result = None
+    for result in self.log_messages:
+      status, message, _ = result
+      if status in check_statuses:
+        last_result = result
+      if status == ENDCHECK:
+        assert last_result # The check must have emitted
+                           # at least one log message!
+
+        # we always test at least the status:
+        assert last_result[0] == expected_status
+
+        if expected_message: # and testing the message is optional
+          assert last_result[1] == expected_message
+        break
+
+
+class FBCheckTesterBadParamsError(Exception):
+  pass
+
+
+class FBCheck_Tester():
+
+  def __init__(self, profile, checkid):
+    self.profile = profile
+    self.checkid = checkid
+    self.log_messages = None
+    self.testcase_number = 0
+    print(f"Testing check '{checkid}' from the '{profile}' profile.")
+
+  def run(self, params, reasoning):
```

Comment:
when we call this method `__call__` we get:

```py
check = FBCheck_Tester("googlefonts",
                         "com.google.fonts/check/001")
check(filename,
              f"Canonical name '{filename}'").assert_result(PASS)

```
instead of

```py
check.run(filename,
              f"Canonical name '{filename}'").assert_result(PASS)
```